### DISCUSSION TITLE
BE COOL BC-B2500 Balkonspeicher 2,24kWh Blutooth

### DISCUSSION THREAD

**Discussion by mastertape** [hoylabs/OpenDTU-OnBattery#553] [Ideas]
Hallo wäre es möglich Akku der über Bluetooth steuerbar und auslesbar ist über die vorhandene esp-32 schnitstelle des esp32 anzubinden?

Beim Be Cool gäbe es eine dokumentierte BT Api, Es geht wogl irgebdwann auch mqtt die aber wohl einen amazon awz dev acoount voraus und ne clod web api abfrage...

Ich habe die Dokumente da soo sie aber nicht offen ins netz stellen, falls jemand versiertes mal drüber gucken möchte würde ich es gerne zukommen lassen.

Der Akku an sich ist sehr interessant ich habe 2 die dann an einen hms 1600 laufen werden.

**Comment by schlimmchen** [comment]
Über die Integration von Bluetooth hatte ich schon länger nachgedacht und es auch kurz probiert. Dabei kam heraus, das die Menge an Code, die für den Bluetooth-Stack hinzukommt, die Größe der Firmware sprengt. Dazu gab es auch eine kurze Diskussion, die ich aber leider nicht mehr finden kann. Die ist irgendwo vergraben... Jedenfalls ist unabhängig von der konkreten Anwendung (z.B. Victron SmartSolar ansprechen) die Nutzung der Bluetooth-Funktionalität in OpenDTU sehr schwierig bzw. nur auf Geräte mit mehr Flash umsetzbar ohne essentielle andere Features zu opfern...

**Reply by Manos1966** [reply]
> die ich aber leider nicht mehr finden kann

Ich tippe auf #468

**Reply by mastertape** [reply]
Dank interessanter Thread - ich habe ein fusion board die s3 eps´s sind meines wissens flah mäsig was üppiger ausgetattet oder?
wenn man sich selber ein image backt und dachen raus lässt die man nicht benötigt solle es gehen.. 

Tasmota z.B. hat wegen dieser Problematik eine schöne Lösung wie man online ne firmware backen kann und diverse "module" zusammen klicken kann... https://github.com/benzino77/tasmocompiler - müsste doch eigentlich mit ein paar anpassungen auch für open dtu nutzbar sein...

**Comment by mastertape** [comment]
Der Akku hat auch noch 2 andere API´s einmal über einen Cloudserver wo wohl in 10-15- min neue daten zur Verfügung stehen  und eine mqtt Api die wohl über einen amazon s3 server läuft. Angeblich ist pv-selbstbau schon dabei es über ihre smart micro controller zu integrieren... Der ist leide nicht Opensource

Der Akku ist übrigens noch unter 2 anderen Namen zu haben Blue Palms (Baumarkt Marke) und Greensolar.de bei Green solar wird das ding erst im März lieferbar sein. In der FAQ von Greensolar wird was von mqtt steuerung erwähnt.

Von meinen Verkäufer war jemand vom tehnischen Support mir 3 PDF´s mit den Schittstellen beschreinungen zu kommen zu lassen wenn jemand der mehr Ahnung hat wie ich mal drüber gucken möchte würde ich sie zukommen lassen

**Reply by redxeagle** [reply]
Ich würde mir die API mal anschauen wollen. Ich habe einen B2500 von GreenSolar. Aber die sind alle identisch.

**Reply by mastertape** [reply]
Wo kann ich dir die zukommen lassen ohne es hier öffentlich einzustellen?
(erledigt)

**Reply by RoFranz** [reply]
Ist in dieser auch zu erkennen, wie man den mqtt server von amazon abfragt? Dann würde ich mir diese auch gerne anschauen...

**Reply by bastian2511** [reply]
@mastertape @redxeagle  Besteht die Möglichkeit an die Unterlagen zu kommen? Habt ihr schon eine MQTT Anbindung umgesetzt bekommen? Ich habe auch den B2500 von Greensolar. Dank im Voraus für eure Hilfe.

**Reply by mastertape** [reply]
alles was aus den unterlagen hervorgeht - ist nicht besonders interessant und überarbeitet @noone2k sowie ein anderes projekt leisten ganze arbeit
https://gist.github.com/noone2k/2ddea4c9bf116aaaefb8626b064d9a41

**Reply by bastian2511** [reply]
![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/80447497/b7eb5a11-d34a-4781-8114-41bd0cacf6f4)

@mastertape  @noone2k  Hat vielleicht jemand von euch eine Idee - was hier falsch läuft?

**Comment by noone2k** [comment]
mehr oder weniger durch zufall auf diesen thread gestossen, wurde durch irgendwas mit ner notification auf dieses project aufmerksam.

hier mal mein ansatz m.h. von esphome ... das eine oder andere könnte hilfreich sein: https://gist.github.com/noone2k/2ddea4c9bf116aaaefb8626b064d9a41

**Comment by mastertape** [comment]
Sry für meine späte Antwort habe die Tage meine Panels - Verkabelt usw.  aufs dach gebracht... 
Ich verfolge deine Scripte mit Spannung. Kann ich dir irgendwie die API Datenblätter die ich zugespielt bekommen habe zukommen lassen? (Mail im folgecoment  und wieder löschen) ?

LG
p.s. Sobald ich die Anlage die Tage am laufen habe wollte ich mich dem Akku widmen

**Comment by Leviathan09** [comment]
Also ich habe so einen B2500 Akku hier stehen...noch

Von dem Akku würde ich die Finger lassen, meiner macht selbst mit neuer Firmware nur Probleme.
Ist die meiste Zeit weder per WLAN noch Bluetooth erreichbar.
Selbst über das ESPHome Projekt was über Bluetooth mit dem Akku kommuniziert funktioniert es nicht wirklich reibunglos.

Dazu kommt das die V1 des Akkus die vom Hersteller versprochene MQTT Fähigkeit nicht unterstütz und diese Funktionalität erst in V2 oder V3 möglich sein wird. Diese Version wird es vermutlich erst gegen März geben.
Alle aktuell ausgelieferten Akkus sind noch V1.

**Reply by mastertape** [reply]
also auf einen anderen Github Projekt von noone2k tut sich einiges so ist der akku per esp-32 mittlerweile zu 90% komplett ausgelesen und steuerbar ... der akku bleibt schon interessant und wird mittlerweile auch bei Hornbach verkauft  was ja keine kleine klitsche ist... Der importeur arbeitet hart an den noch bestehenden problemen vieles wurde in den verganfenen wochen gelöst.... Wäre klasse wenn jemand der hier aktiv mit coded  vielleicht mal auf das wachsende projekt von noone2k (der eine super arbeit macht!) sich anguckt oder nen interssierten blick drauf wirft... 

eigentlich solle ähnlich wie bei tasmota powermetern auch -  ne möglichkeitda sein wichtige interkationen und datenübertragungen z,b, wie auf passthrough schalten ladeszustände usw zu realisieren. 

Ich glaube das noch einige diesen akku kaufen werden und hier drüber stolpern

**Comment by mastertape** [comment]
Das Fiat von noon2k hat einen kommentar Bereich… wäre vielleicht die
passendere stelle da sind mehr Leute als hier

**Comment by redxeagle** [comment]
Ich kann mastertape nur zustimmen. Aus den Unterlagen geht nicht viel hervor.

Laut Aussage von Greensolar wird die V1 MQTT Unterstützung erhalten. 
Laut Hersteller ist heute die Firmware 134 auch released wurden. Ich war aber noch nicht zuhause um es einzuspielen.

Wir warten es mal ab.

Aus den Unterlagen ging aber auch hervor das hier meines Erachtens auch ein Sicherheitsleck vorliegt. Dem nehmen ich mich auch noch einmal an. Die API und Co Richtung Cloud ist nicht gesichert und funktioniert nur via HTTP. So ganz glücklich bin ich hier nicht. 
Ich hoffe das mit MQTT das dann besser wird.

Die ESP Variante von noon2k habe ich mir auch noch nicht vorgenommen und hoffe das einfacher direkt mit IOBroker umsetzen zu können.

Besten Gruß
Daniel

**Reply by bastian2511** [reply]
Hey Daniel. Konntest du die neue Firmware testen? Wie bist du darauf aufmerksam geworden?

**Comment by mxgross** [comment]
Gibts hier mittlerweile was neues bzgl. MQTT Zugriff oder wie man sonst irgendwie lesend ein paar Daten vom Akku abgreifen kann?
Habe die Hornbach Bluepalm BP-B2500 V 1.2 Variante mit Firmware V157, Kommunikationsmodul 20231023xxxx.

**Reply by muenstermann1** [reply]
Wenn du die V1 hast musst du dem Support des Akkus schreiben das du MQTT freischalten möchtest.
Danach ist aber der Cloud Support futsch und du kannst nur noch lokal direkt drauf zugreifen oder halt über den MQTT Broker.

**Reply by mxgross** [reply]
Danke, ich hatte vor ein paar Wochen eine URL fürs Webportal bekommen und MQTT aktiviert, wollte dann aber doch nicht auf die App verzichten und habe es wieder deaktiviert. Seitdem habe ich trotzdem nur noch via Bluetooth aktuelle Daten in der Power Zero App. Der Support wollte sich vor einer Woche via Mail zurückmelden, aber ich habe nichts mehr gehört.

**Reply by 3joern** [reply]
Habe die Tage auch rumgespielt, Du kannst einfach die Werkseinstellungen (kein Hardware Reset) über die App Settings mit Bluetooth Verbindung wiederherstellen (wenn Du neben dem Speicher stehst und BT grün ist), dann App neu starten und in den Einstellungen WLAN neu einrichten. Nicht vergessen, Deinen DOD und die Schwelle an Deine Bedürfnisse wieder anzupassen. Habe das inzwischen 2x mal so gemacht und der Speicher läuft genau so wie er soll. 
Wenn Ihr mehr über das MQTT wissen wollt empfehle ich den Thead beim [Photovoltaikforum ](https://www.photovoltaikforum.com/thread/232408-b2500-bluepalm-be-cool-marstek-etc-mqtt-aktivierung-und-smarthome-einbindung)
Bin am experimentieren, das mit dem ioBroker und dessen MQTT auszulesen, klappt irgendwie aber das abholen und senden über Java mit Hilfe von ChatGPT (weil ich ein absoluter noob bin) ist .... ausbaufähig.

**Reply by mastertape** [reply]
guck dir das git von noon2k an da istdie komplette kommunikation dokumentiert es gibt ein gist und ein git von noon2k
aus der gist diskussion geht hervor das man betreiber schuss in österreich sich mqtt direkt freischalzten lassen kann - das geht neuedings dann geht aber die app nicht mehr dafür kann man lokal allles per mqtt regeln
